### PR TITLE
Using cache reference for query UI menu

### PR DESCRIPTION
**Pull request by mrproliu**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java [hunk 1]
```diff
@@ -35,25 +40,29 @@
...
-    private boolean isMenuItemsBeenFetched = false;
     private MetadataQueryService metadataQueryService;
 
-    public UIMenuManagementService(ModuleManager moduleManager) {
+    public UIMenuManagementService(ModuleManager moduleManager, CoreModuleConfig moduleConfig) {
         this.moduleManager = moduleManager;
-        this.menuItemFuture = new CompletableFuture<>();
+        this.menuItemCache = CacheBuilder.newBuilder()
+            .maximumSize(1)
+            .refreshAfterWrite(moduleConfig.getUiMenuRefreshInterval(), TimeUnit.SECONDS)
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java:59]: RefreshAfterWrite? Where would we write this cache?
Do I misread the codes?

Comment (mrproliu) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java:59]: The cache would be read from the database and written into the cache when accessing the menu for the first time. After the writing is completed, a timer starts counting, and when it reaches a certain point, the data would be automatically rewritten.

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java:59]: Why push first time late? This kind of mechanism usually is used for cold data, but menu is super hot. I think the timer should be up and running after initialized(if needed).

Comment (mrproliu) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java:59]: updated, please re-check

**Review by wu-sheng — DISMISSED** [review]
@kezhenxu94 Another round check?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java [hunk 2]
```diff
@@ -83,46 +92,23 @@ public void saveMenuAndStartFetch(List<UIMenuItemSetting> menuItems, int fetchIn
         menu.setConfigurationJson(GSON.toJson(menuItems));
 
         getMenuDAO().saveMenu(menu);
-        startingFetchMenu(fetchInterval);
-    }
-
-    private void startingFetchMenu(int fetchIntervalSecond) {
-        Executors.newSingleThreadScheduledExecutor()
-            .scheduleWithFixedDelay(this, 0, fetchIntervalSecond, java.util.concurrent.TimeUnit.SECONDS);
+        this.menuItemCache.put(true, fetchMenuItems());
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/management/ui/menu/UIMenuManagementService.java:95]: There is no point to call cache `fetchMenuItems`, just call cache.get(true) and the loading cache finds it absent and will call the fetchMenuItems inside the load method

```suggestion
        this.menuItemCache.get(true);
```

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**